src/main/java/pl/javastart/task/Group.java

```diff
@@ -0,0 +1,94 @@
+package pl.javastart.task;
+
+class Group {
+    private final String code;
+    private final String name;
+    private Lecturer lecturer;
+    private int currentStudentsNumber = 0;
+    private Student[] students;
+    private double[] grades;
+
+    Group(String code, String name, Lecturer lecturer) {
+        this.code = code;
+        this.name = name;
+        this.lecturer = lecturer;
+        students = new Student[University.STUDENTS_PER_GROUP];
+        grades = new double[University.STUDENTS_PER_GROUP];
+    }
+
+    void printGrades() {
+        for (int i = 0; i < students.length; i++) {
+            if (grades[i] != 0) {
+                System.out.println(students[i].getInfo() + ": " + grades[i]);
+            }
+        }
+    }
+
+    void printGroupInfo() {
+        System.out.printf("Kod: %s%nNazwa: %s%nProwadzący: %s%nUczestnicy:%n%s",
+                code, name, lecturer.getInfo(), getStudentsInfo());
+    }
+
+    private String getStudentsInfo() {
+        StringBuilder list = new StringBuilder();
+        Student student;
+
+        for (int i = 0; i < currentStudentsNumber; i++) {
+            student = students[i];
+            list.append(student.getIndex() + " " + student.getFirstName() + " " + student.getLastName() + "\n");
+        }
+
+        return list.toString();
+    }
+
+    void addStudent(Student student) {
+        if (currentStudentsNumber >= University.STUDENTS_PER_GROUP) {
+            extendStudentsAndGrades();
+        }
+
+        students[currentStudentsNumber++] = student;
+    }
+
+    void printStudentGrade(int studentPositionInGroup) {
+        System.out.println(name + ": " + grades[studentPositionInGroup]);
+    }
```

Comment:
Unikaj dodawania wydruków w swoich klasach domenowych. Przez printlny przywiązujesz tę klasę do działania w konsoli. Jeżeil zamianisz ją na taką, która zwróci Stringa, to bez problemu można jej użyć np. w aplikacji z GUI, czy aplikacji webowej